ArchUnitNET/Domain/PlantUml/PlantUmlParser.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Security;
+using System.Text;
+using System.Text.RegularExpressions;
+using static ArchUnitNET.Domain.PlantUml.PlantUmlPatterns;
+
+namespace ArchUnitNET.Domain.PlantUml
+{
+    internal class PlantUmlParser
+    {
+        private PlantUmlPatterns _plantUmlPatterns = new PlantUmlPatterns();
+        public PlantUmlDiagram Parse(string filename)
```

Comment:
we could additionally allow for streams. That could make it easier to test against architecture documentations, but more importantly: we could simplify some test code, which would not need to write files, but the test files could be inserted as an in memory stream

Reply:
Good idea.